pkg/eestream/encryption.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package eestream
+
+import (
+	"github.com/zeebo/errs"
+)
+
+// Cipher is a type used to define the type of encryption to use
+type Cipher byte
+
+// Constant definitions for no encryption (0), AESGCM (1), and SecretBox (2)
+const (
+	None = Cipher(iota)
+	AESGCM
+	SecretBox
+)
+
+// Constant definitions for key and nonce sizes
+const (
+	GenericKeySize   = 32
+	GenericNonceSize = 24
+	AESGCMNonceSize  = 12
+)
+
+// GenericKey represents the largest key used by any encryption protocol
+type GenericKey [GenericKeySize]byte
+
+// Bytes returns the key as a byte array pointer
+func (key *GenericKey) Bytes() *[GenericKeySize]byte {
```

Comment:
Is there a reason it's returning the byte array pointer, rather than a byte slice or byte array? Just wondering?